File: LeetCode/cpp/526.cpp
class Solution {
public:
    int countArrangement(int N) {
        int num = 0;
        vector<int> used(N + 1, 0);
        BackTracking(1, num, used, N);
        return num;
    }

    void BackTracking(int n, int &num, vector<int> &used, int &N) {
        if (n > N) {
            ++num;
            return;
        }
        for (int i = 1; i <= N; ++i) {
            if (used[i] == 0 && (i % n == 0 || n % i == 0)) {
                used[i] = 1;
                BackTracking(n + 1, num, used, N);
                used[i] = 0;
            }
        }
    }
};

File: LeetCode/python3/526.py
class Solution:
    def countArrangement(self, N: int) -> int:
        visited = [False for i in range(N)]
        self.count = 0

        def Count(k, visited):
            if k == N:
                self.count += 1
            for i in range(N):
                if not visited[i] and ((k + 1) % (i + 1) == 0 or (i + 1) % (k + 1) == 0):
                    visited[i] = True
                    Count(k + 1, visited)
                    visited[i] = False

        Count(0, visited)
        return self.count
